plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2938,8 +2998,12 @@ void producer_plugin_impl::produce_block() {
    _time_tracker.clear();
 }
 
-void producer_plugin::received_block(uint32_t block_num) {
+void producer_plugin::received_block(uint32_t block_num, chain::fork_db_add_t fork_db_add_result) {
    my->_received_block = block_num;
+   if (fork_db_add_result == fork_db_add_t::fork_switch) {
```

Comment:
Add a comment saying fork_db_add_t::fork_switch means head block of best fork (different from the current branch) is received. Or just pass in a parameter of new_best_fork instead of fork_db_add_result

Reply:
https://github.com/AntelopeIO/spring/pull/1047/commits/1efb62c7899f3997fe4b4836b0f12939fb63da47